### PR TITLE
Add tests for urEnqueueUSMFill2D

### LINKED ISSUE #240: Adds CTS tests for `urEnqueueUSMFill2D` entry-point
We should add conformance tests for `urEnqueueUSMFill2D`.

Required Coverage:
* At least one successful (i.e. called with correct arguments and assert on expected result, not applicable to `CreateWithNativeHandle` entry points)
* If this a query entry-point or one which accepts flags etc : we should tests every possible enumeration (where possible).
* A test case for each returned error code (exclude disaster situations i.e. OUT_OF_MEMORY etc)

### PR DESCRIPTION
**Pull request by fabiomestre**
Closes #240

### CODE REVIEW

File: test/conformance/enqueue/urEnqueueUSMFill2D.cpp [hunk 1]
```diff
@@ -1,2 +1,288 @@
...
+
+    void TearDown() override {
+        if (ptr) {
+            EXPECT_SUCCESS(urUSMFree(context, ptr));
+        }
+
+        UUR_RETURN_ON_FATAL_FAILURE(uur::urQueueTest::TearDown());
+    }
+
+    const size_t pitch = 16;
```
Comment (veselypeta) [inline test/conformance/enqueue/urEnqueueUSMFill2D.cpp:185]: I would avoid making member variables `const` : [see here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-constref)

Comment (veselypeta) [reply test/conformance/enqueue/urEnqueueUSMFill2D.cpp:185]: make it `static constexpr` instead

**Review by veselypeta — COMMENTED** [review]
Looks good. Just some suggestions.

File: test/conformance/enqueue/urEnqueueUSMFill2D.cpp [hunk 2]
```diff
@@ -1,2 +1,288 @@
...
+
+struct urEnqueueUSMFill2DTestWithParam
+    : uur::urQueueTestWithParam<testParametersFill2D> {
+    void SetUp() override {
+        UUR_RETURN_ON_FATAL_FAILURE(urQueueTestWithParam::SetUp());
+
+        pitch = std::get<1>(GetParam()).pitch;
+        width = std::get<1>(GetParam()).width;
+        height = std::get<1>(GetParam()).height;
+        pattern_size = std::get<1>(GetParam()).pattern_size;
```
[the remark this answers is not in the document]
Comment (fabiomestre) [reply test/conformance/enqueue/urEnqueueUSMFill2D.cpp:36]: Thanks for the tip. I think there is a bit too much going on with the structured bindings approach (create 4 new variables to then make a tuple and finally assign that tuple to the variable).  So I prefer the original approach.

**Review by veselypeta — APPROVED** [review]
LGTM

**Review by martygrant — APPROVED** [review]
LGTM

**Review by kbenzie — APPROVED** [review]
LGTM

**Pull request merged**